### ISSUE TITLE
Pruning only on some models?

### ISSUE THREAD

**Issue by fickas** [optuna/optuna-examples#229]
I do not see how I can define a pruner for something like logistic regression. I do not see what the `step` value would be. I currently set it to 1 which in essence does no pruning I believe.

Also, am I correct that I can tune KNN by using `n_neighbors` as the step?

**Comment by nzw0301** [comment]
> I do not see how I can define a pruner for something like logistic regression. I do not see what the step value would be. I currently set it to 1 which in essence does no pruning I believe.

In an iterative algorithm like gradient descent, it would be iteration or step is `step` and `value` is validation loss or some metric which should have consistent direction with the objective value. 


> Also, am I correct that I can tune KNN by using n_neighbors as the step?

It sounds wrong.

**Comment by nzw0301** [comment]
Anyway, this question is not related to this repo. Please use https://github.com/optuna/optuna/discussions to ask questions about optuna.
